### ISSUE TITLE
staticFileGlobs with "non-present-right-now" files 

### ISSUE THREAD

**Issue by acanimal** [GoogleChrome/sw-precache#230]
Hi,
it can sound a bit weird at the beginning but let me explain :)

I'm using sw-precache through the sw-precache-webpack-plugin project. I'm using webpack-dev-middleware too, and because of this my `bundle.js` and `style.css` files really not exists while developing but they are generated on the fly by the webpack-dev-middleware.

Although I specify my JS and CSS files in the `staticFileGlobs` property they are not included in the generated SW code. I debug the sw-precache-webpack-plugin code and it doesn't seems a limitation of the plugin but the sw-precache.

I'm right? Is there any restriction on the values of the `staticFileGlobs` property to accept only "existent" files? If so, is there any option to bypass this? I want that files in the generated code so the SW once installed precache that files which, although not strictly exists in disk, can be generated by the server.

Thanks in advance.

**Comment by goldhand** [comment]
@acanimal this is related to sw-precache-webpack-plugin, sw-precache is working correctly. The issue you reference above is the right area for this question: https://github.com/goldhand/sw-precache-webpack-plugin/issues/17

**Comment by goldhand** [comment]
The webpack in-memory generated `bundle.js` and `style.css` files served from webpack-dev-middleware will not have the same name or path as the physical files.

1. A service worker file will need to be generated by sw-precache.
2. This service worker will need to be registered by a script in your html.
3. The service worker will implement it's caching strategy using the options you specify.

This is difficult because you can't specify the name of the webpack-dev-middleware generated files before you generate the service worker because the webpack-dev-middleware generated files don't exist yet.

A crappy solution might be using a glob like `**/*.js` in `staticFileGlobs`.

I think this will cache a lot of files though, every time you make a change a new asset will be emitted. 

That's why I think it would be better to do something with webpack.

sw-precache has a generate method that we can use to make the service worker in memory. If we can emit the service-worker with webpack-dev-middleware maybe we can use the webpack stats with the service worker there. Can continue to discuss goldhand/sw-precache-webpack-plugin#17

**Comment by acanimal** [comment]
Not exactly. In my case at least, while developing and using webpack-dev-middleware, I know the exactly names, path and url where to "found" the generated js and css files.

Didn't look much in depth but seems sw-precache filters the files specified in `staticFileGlobs` that does not exists on disk (https://github.com/GoogleChrome/sw-precache/blob/master/lib/sw-precache.js#L157)

My idea is to use that or another new property so I can specify a file name/path to be added to the final list of files to be cached without the need that those files currently exists.
